feat(core): Make ConfigModule importable in every NgModule

UserGalileo opened the pull request:
Removed the check in the constructor, this way every module can import ConfigModule and have access to ConfigPipe.
Updated ReadME with a one-liner instruction for ConfigPipe.

** PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/fulls1z3/ngx-config/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

** PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

** What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #149 

** What is the new behavior?

Now every NgModule can import ConfigModule.

** Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

** Other information

I don't think tests are needed for this change: can you confirm?

UserGalileo committed:
Updated config.module.ts

Removed the check in the constructor, this way every module can import ConfigModule and have access to ConfigPipe.

UserGalileo committed:
Update README.md

fulls1z3 reviewed (changes requested):
i appreciate the efforts to make `ConfigPipe` available in every module, but I'd recommend to keep the root check and come up with `forChildren` method.

this way we can both preserve exiting functionality and add new ones.

UserGalileo commented:
@fulls1z3 Can you confirm that the method should be named `forChildren`? I don't think I've ever seen a `forChildren` method in other libraries, I'd suggest to choose between these 2 options:
- `forFeature` a-la NgRx
- `forChild` a-la Angular Router

But if you choose `forChildren` no problem, I just want to be sure before making new commits :)

Also, for the root check: it's doable but we'd have to perform the check outside of the constructor, just like `Angular Router` does: [this way.
](https://github.com/angular/angular/blob/master/packages/router/src/router_module.ts#L197)
Do you agree with adopting this solution?

fulls1z3 commented:
@UserGalileo I agree and it's a typo - let's follow the Angular pattern and call it `forChild`.

fulls1z3 commented:
@UserGalileo also 👍 for the root guard

UserGalileo committed:
Provide a forChild() method

This change makes ConfigModule importable in lazy loaded modules with forChild() but also preserves the forRoot() check.

UserGalileo committed:
Updated README to reflect last commit

UserGalileo commented:
@fulls1z3 I made the changes on my branch, could you verify if everything works fine?

I'm sorry, I would be happy to provide some more tests for the lazy-loaded modules scenario, but I'm really busy these days.

fulls1z3 committed:
Merge branch 'master' into patch-1

fulls1z3 commented:
@UserGalileo just fixed the CI pipeline, but it turns out your PR has to pass validation checks such as TSLint.

FYI, you can add the following block to `tslint.json` between lines 4 and 5 - in order that `void` type works for `InjectionToken`.

```

    "invalid-void": [
      true,
      {
        "allow-generics": [
          "EventEmitter",
          "InjectionToken",
          "Observable",
          "Promise"
        ]
      }
    ],
```

fulls1z3 commented on packages/@ngx-config/core/src/config.module.ts:
```diff
@@ -13,17 +13,37 @@ export const initializerFactory = (config: ConfigService) => {
+
+/**
+ * Used to check if the user has called ConfigModule.forRoot() in a lazy-loaded module.
+ */
```
your code is already good - explaining itself. no need for comments ;)

fulls1z3 commented on packages/@ngx-config/core/src/config.module.ts:
```diff
@@ -13,17 +13,37 @@ export const initializerFactory = (config: ConfigService) => {
+  /**
+   * By injecting the guard we're making sure that the check will be made every time
+   * a new ConfigModule is imported, both in the root injector and in lazy loaded modules.
+   */
```
same as above - your code is already crystal clear

fulls1z3 reviewed (changes requested):
all clear, even more 🙂

fulls1z3 committed:
Merge branch 'master' into patch-1

UserGalileo commented:
Perfect, is there anything left to do on my side? :)

fulls1z3 commented:
yeah, you could follow the code review (removing comments from the code) as well as Sider must return no issues (tslint). Please consider my remarks at https://github.com/fulls1z3/ngx-config/pull/157#issuecomment-560753719 for this.

miguelchico commented:
Hi @UserGalileo, @fulls1z3 do you expect any progress on this PR?

UserGalileo committed:
Update config.module.ts

Removed the generic from InjectionToken

UserGalileo committed:
Update config.module.ts (tslint)

UserGalileo commented:
Sorry for the delay, these are really busy months for me. I just had the time to fix the tslint errors, it seems fine now, could you check?

I preferred to disable tslint rules for 2 lines of code instead of messing around with the tslint.json file, since there were only 2 occasions, and I removed the generic from `InjectionToken` since it was useless.

fulls1z3 committed:
Merge branch 'master' into patch-1
